### PR TITLE
Adds Reader.timeseries

### PR DESCRIPTION
**Pull request by richardjgowers**
Fixes partially #1383

Changes made in this Pull Request:
 - added deprecation warnings to all `core.Timeseries` classes
 - added `timeseries` method to ProtoReader

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

**Comment by richardjgowers** [comment]
So Encore uses `DCDReader.timeseries` (which isn't actually a `Timeseries` (as in the class)) but instead just a handy method for iterating and grabbing coordinates.  I've put a implementation of this into `ProtoReader` so that all readers get a basic version of it

File: package/MDAnalysis/coordinates/base.py [hunk 1]
```diff
@@ -1461,6 +1461,60 @@ def __repr__(self):
             natoms=self.n_atoms
         ))
 
+    def timeseries(self, asel=None, start=None, stop=None, step=None,
```
Comment (orbeckst) [inline package/MDAnalysis/coordinates/base.py]: This is a new feature for trajectory readers. It cannot go into a patch release. It has to be in 0.17.0.

File: package/MDAnalysis/coordinates/DCD.py [hunk 2]
```diff
@@ -578,6 +579,7 @@ def timeseries(self, asel=None, start=None, stop=None, step=None, skip=None,
         # XXX needs to be implemented
         return self._read_timeseries(atom_numbers, start, stop, step, format)
 
+    @deprecate(message="This method will be removed in 0.17")
```
Comment (orbeckst) [inline package/MDAnalysis/coordinates/DCD.py]: All the deprecations can go in now. 

I somehow missed your PR and did the reST deprecations in PR #1403

**Comment by orbeckst** [comment]
@richardjgowers I cherry-picked your first two commits onto #1403

File: package/MDAnalysis/coordinates/base.py (hunk 1 above)
Comment (richardjgowers) [inline package/MDAnalysis/coordinates/base.py]: Ok, I'll bump this into 0.17 then

File: package/CHANGELOG [hunk 3]
```diff
@@ -17,6 +17,9 @@ mm/dd/yy
 
   * 0.16.2
 
+Deprecations
+  * deprecated core.Timeseries module (Issue #1383)
```
Comment (kain88-de) [inline package/CHANGELOG]: this needs to be moved up the timeline or removed right?

File: package/MDAnalysis/coordinates/base.py [hunk 4]
```diff
@@ -1461,6 +1461,60 @@ def __repr__(self):
...
+            raise NotImplementedError
+
+        # allocate output array
+        coordinates = np.empty((nframes, natoms, 3), dtype=np.float32)
+        for i, ts in enumerate(self[start:stop:step]):
+            # if atom_number == None, this will cause view of array
+            # do we need copy in this case?
+            coordinates[i] = ts.positions[atom_numbers].copy()
+
+        return coordinates
```
Comment (kain88-de) [inline package/MDAnalysis/coordinates/base.py]: This is not implementing the `format` keyword argument. I guess you can copy the code I use in libdcd for this. There are also a lot of tests for it right now.

File: package/CHANGELOG (hunk 3 above)
Comment (richardjgowers) [inline package/CHANGELOG]: Yeah it's leftover from when this also did the deprecation

File: package/MDAnalysis/coordinates/base.py (hunk 4 above)
Comment (richardjgowers) [inline package/MDAnalysis/coordinates/base.py]: And yeah, I'll have to deal with that in this PR too

File: package/MDAnalysis/coordinates/base.py [hunk 5]
```diff
@@ -1461,6 +1461,60 @@ def __repr__(self):
...
+             Begin reading the trajectory at frame index `start` (where 0 is the index
+             of the first frame in the trajectory); the default ``None`` starts
+             at the beginning.
+        stop : int (optional)
+             End reading the trajectory at frame index `stop`-1, i.e, `stop` is excluded.
+             The trajectory is read to the end with the default ``None``.
+        step : int (optional)
+             Step size for reading; the default ``None`` is equivalent to 1 and means to
+             read every frame.
+        format : str (optional)
```
Comment (kain88-de) [inline package/MDAnalysis/coordinates/base.py]: btw I'm for naming this option `order` format is confusing with the buildin

**Comment by orbeckst** [comment]
> Am Jun 25, 2017 um 06:07 schrieb Max Linke <notifications@github.com>:
> 
> 
> +             read every frame.
> +        format : str (optional)
> btw I'm for naming this option order format is confusing with the builtin
> 
I think we already made this decision for the MemoryReader and said we would change Timeseries later. 

(Typing from mobile so difficult to chase down issues to reference ... apologies)

File: package/MDAnalysis/coordinates/base.py (hunk 4 above)
Comment (orbeckst) [inline package/MDAnalysis/coordinates/base.py]: And we wanted to change the `format` kwarg to `order`.

File: package/MDAnalysis/coordinates/base.py (hunk 5 above)
Comment (orbeckst) [inline package/MDAnalysis/coordinates/base.py]: See #1063 where this was already mooted.

Comment (orbeckst) [inline package/MDAnalysis/coordinates/base.py]: See #1453 for the issue.

**Pull request closed without merging**